src/base/bittorrent/torrentcreator.cpp

```diff
@@ -205,13 +206,27 @@ void TorrentCreator::run()
 
         checkInterruptionRequested();
 
-        // create the torrent
-        const nonstd::expected<void, QString> result = Utils::IO::saveToFile(m_params.savePath, entry);
+        const auto result = std::invoke([torrentFilePath = m_params.torrentFilePath, entry]() -> nonstd::expected<Path, QString>
+        {
+            if (!torrentFilePath.isValid())
+                return Utils::IO::saveToTempFile(entry);
+
+            const nonstd::expected<void, QString> result = Utils::IO::saveToFile(torrentFilePath, entry);
+            if (!result)
+                return nonstd::make_unexpected(result.error());
+
+            return torrentFilePath;
+        });
         if (!result)
             throw RuntimeError(result.error());
 
-        emit updateProgress(100);
-        emit creationSuccess(m_params.savePath, parentPath);
+        BitTorrent::TorrentCreatorResult creatorResult;
+        creatorResult.torrentFilePath = result.value();
+        creatorResult.savePath = parentPath;
+        creatorResult.pieceSize = newTorrent.piece_length();
```

Comment:
IMO the designated initializers syntax is nice and it also avoid duplicate initialization. Maybe you should consider using it:
```suggestion
        const BitTorrent::TorrentCreatorResult creatorResult
        {
            .torrentFilePath = result.value(),
            .savePath = parentPath,
            .pieceSize = newTorrent.piece_length()
        };
```